cmd/bcpd/app/msg.go

```diff
@@ -0,0 +1,55 @@
+package app
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/x/batch"
+)
+
+var _ batch.Msg = (*BatchMsg)(nil)
+
+func (*BatchMsg) Path() string {
+	return batch.PathExecuteBatchMsg
+}
+
+func (msg *BatchMsg) MsgList() ([]weave.Msg, error) {
+	messages := make([]weave.Msg, len(msg.Messages))
+	// make sure to cover all messages defined in protobuf
+	for i, m := range msg.Messages {
+		res, err := func() (weave.Msg, error) {
```

Comment:
You don't need anonymous function, you can type cast without it.

Reply:
Also it is done because there is a loop and two return params, which makes things easier if they are wrapped like this.